### PR TITLE
feat: Add secure CloudAccount resource

### PR DESCRIPTION
**Pull request by nkraemer-sysdig**
Add a `CloudAccount` resource to the Sysdig Secure provider

### CODE REVIEW

File: sysdig/internal/client/secure/models.go
Comment (gautam417) [inline sysdig/internal/client/secure/models.go:363]: @nkraemer-sysdig Do we want to make this one `externalId` that comes back from the CloudAccount response?

**Review by gautam417 — APPROVED** [review]

**Comment by tembleking** [comment]
Thanks for the PR @nkraemer-sysdig! The tests are failing currently with 
```
=== CONT  TestAccSecureCloudAccount
    resource_sysdig_secure_cloud_account_test.go:18: Step 1/3 error: Error running apply: exit status 1
        
        Error: Post "https://secure.sysdig.com/api/cloud/v2/accounts?includeExternalID": POST https://secure.sysdig.com/api/cloud/v2/accounts?includeExternalID giving up after 5 attempt(s)
        
          with sysdig_secure_cloud_account.sample,
          on terraform_plugin_test.tf line 2, in resource "sysdig_secure_cloud_account" "sample":
           2: resource "sysdig_secure_cloud_account" "sample" {
        
--- FAIL: TestAccSecureCloudAccount (16.21s)
```

May it be probably because this hasn't been merged to SaaS yet? If so, we may need to wait until this has been merged so we can add it to the provider.

**Comment by nkraemer-sysdig** [comment]
> May it be probably because this hasn't been merged to SaaS yet?

Yep, changes have been deployed to staging but not yet to prod. Will update once changes are available in prod. 

There will also be some minor changes to this PR, there were a couple issues I found while testing, will push fixes shortly. Thanks for taking a look!

**Comment by nestorsalceda** [comment]
Duplicated from #112 to run the tests and are passing.

Merging!

Thank you so much @nkraemer-sysdig and @gautam417

**Pull request merged**
